src/core/events/handler.py

```diff
@@ -83,21 +84,23 @@ def publish_event(self, event, caller=None):
     # executes callbacks on dedicated thread as a daemon
     def worker(self):
         while self.running:
-            queue_lock.acquire()
-            hook = self.get()
-            queue_lock.release()
             try:
+                hook = self.get()
                 hook.execute()
             except Exception as ex:
                 logging.debug("Exception: {} - {}".format(hook.__class__, ex))
-            self.task_done()
+            finally:
+                self.task_done()
         logging.debug("closing thread...")
 
     def notifier(self):
         time.sleep(2)
+        # should consider locking on unfinished_tasks
```

Comment:
Yeah, this doesn't look terribly safe to me, but we get away with this because it's only reading, and it's only for debugging purposes. Odd that there's no decent interface in Queue for querying this.